src/bacnet/basic/object/nc.c

```diff
@@ -414,15 +414,17 @@ bool Notification_Class_Write_Property(BACNET_WRITE_PROPERTY_DATA *wp_data)
                 if (len == BACNET_STATUS_REJECT) {
                     wp_data->error_class = ERROR_CLASS_PROPERTY;
                     wp_data->error_code = ERROR_CODE_INVALID_DATA_TYPE;
-                    return false;
+                    status = false;
+                    break;
                 }
                 iOffset += len;
                 /* Increasing element of list */
                 if ((++idx >= NC_MAX_RECIPIENTS) &&
                     (iOffset < wp_data->application_data_len)) {
                     wp_data->error_class = ERROR_CLASS_RESOURCES;
                     wp_data->error_code = ERROR_CODE_NO_SPACE_TO_WRITE_PROPERTY;
-                    return false;
+                    status = false;
+                    break;
                 }
             }
             /* Decoded all recipient list */
```

Comment:
I think you need if (!status) break; here...  the previous break only breaks you out of the while loop, I think you want to break out of the switch case too here